content/en/docs/tasks/security/authorization/authz-ingress/index.md

```diff
@@ -126,7 +126,7 @@ spec:
 
 {{< tab name="Network Load Balancer" category-value="network" >}}
 
-If you are using a TCP/UDP network load balancer that preserves the client IP address (AWS Network Load Balancer, GCP External Network Load Balancer, Azure Load Balancer) or you are using Round-Robin DNS, then you can also preserve the client IP inside Kubernetes by bypassing kube-proxy and preventing it from sending traffic to other nodes.  **However, you must run an ingress gateway pod on every node.** If you don't, then any node that receives traffic and doesn't have an ingress gateway will drop the traffic. See [Source IP for Services with `Type=NodePort`](https://kubernetes.io/docs/tutorials/services/source-ip/#source-ip-for-services-with-type-nodeport)
+If you are using a TCP/UDP network load balancer that preserves the client IP address (AWS Network Load Balancer, GCP External Network Load Balancer, Azure Load Balancer) or you are using Round-Robin DNS, then you can also preserve the client IP inside Kubernetes by bypassing kube-proxy and preventing it from sending traffic to other nodes.  **However, for production deployments it is strongly recommended to deploy an ingress gateway pod on every node** if you enable `externalTrafficPolicy: Local`. Any node that receives NLB ingress traffic and doesn’t have a local ingress gateway to handle it will drop the traffic if `externalTrafficPolicy: Local` is enabled. This creates a situation where **only** nodes with an active ingress gateway pod will be able to accept and distribute incoming NLB traffic to the rest of the cluster, creating potential ingress traffic bottlenecks. See [Source IP for Services with `Type=NodePort`](https://kubernetes.io/docs/tutorials/services/source-ip/#source-ip-for-services-with-type-nodeport)
```

Comment:
LBs health check the nodes though, right? So is the "deploy to all nodes" suggestion so that you don't have downtime while the healthchecks are propogating or something?

Reply:
Yeah lemme reword that, you're correct - the fundamental problem is that with `local` you can only accept LB traffic on the subset of nodes with a gateway, regardless of whether the LB actually tries to send traffic to them or not.

Reply:
sorry, but it's not practical (e.g. several replicas will address SPOF) even suggest to run gateway at every node ):  maybe better to suggest this only if user cannot setup `local`

Reply:
`several replicas will address SPOF` - that would address SPOF of the gateway pod on the node, but not of the node(s), correct?

`it's not practical to even suggest to run gateway at every node` - Agree, it sort of depends on how many nodes you have and what you consider "good enough" for your deployment. The existing docs are making that "every node" claim - if we want to modify that we can, but at the end of the day the answer for "how many" is "it depends on your deployment".

Reply:
AFAIK you can run on a subset of nodes and just have it deployed to all of them there. The issue is only going from N -> 0 pods on a node.

That may be a GCP specific thing though

Reply:
we usualy deploy Gateways that spread across nodes that address node problem

Reply:
@eghobo removed the "all nodes" guidance in favor of "multiple nodes"

Reply:
@bleggett looks good, maybe "it is strongly recommended to deploy an ingress gateway pod on on different nodes"